osdk/src/cli.rs

```diff
@@ -89,12 +90,44 @@ pub struct ForwardedArguments {
 
 #[derive(Debug, Parser)]
 pub struct NewArgs {
-    #[arg(long, default_value = "false", help = "Use the kernel template")]
+    #[arg(
+        id = "type",
+        long = "type",
+        short = 't',
+        default_value = "library",
+        help = "The type of the project to create",
+        conflicts_with_all = ["kernel", "library"],
+    )]
+    pub type_: ProjectType,
```

Comment:
Well, I think we may not need to provide such complicated interface to creating a new project.

For creating a new Rust project with Cargo, we can provide `--bin` and `--lib`, but it does not accept `--type` or `-t`. I think `--kernel` and `--lib` is enough. But setting `--kernel` as the default option may be better. So if a user wants to create a kernel, the user can use the least words.